lib/puppet/provider.rb

```diff
@@ -290,16 +290,26 @@ def self.default_match
     end
   end
 
+  # Compare a fact value against one or more supplied value
+  # @param [Symbol] fact a fact to query to match against one of the given values
+  # @param [Array, Regexp, String] values one or more values to compare to the
+  #   value of the given fact
+  # @return [Boolean] whether or not the fact value matches one of the supplied
+  #   values. Given one or more Regexp instances, fact is compared via the basic
+  #   pattern-matching operator.
   def self.fact_match(fact, values)
-    values = [values] unless values.is_a? Array
-    values.map! { |v| v.to_s.downcase.intern }
-
-    if fval = Facter.value(fact).to_s and fval != ""
-      fval = fval.to_s.downcase.intern
-
-      values.include?(fval)
+    fact_val = Facter.value(fact).to_s.downcase
+    if fact_val.empty?
+      return false
     else
-      false
+      values = [values] unless values.is_a?(Array)
+      values.any? do |value|
+        if value.is_a?(Regexp)
+          fact_val =~ value
```

Comment:
nit: This will return the index in the string where the match took place, not a boolean. Not a biggie, its truthy, though we might want `!!` in front of this, or changing the docs.

Reply:
True, but this is a test inside the Enumerator#any?, which will always return boolean (for the method-level return value).

Reply:
That said, I'm happy to coerce internally to boolean if that makes it more readable

Reply:
ok i tried it and I don't think it makes more readable personally